lib/api.js

```diff
@@ -99,12 +99,12 @@ function apiCall(httpMethod, options, callback) {
 
           apiCall(httpMethod, options, callback);
         } else if (responseBody && responseBody.error) {
-          log(`API Error: ${prettyPrint(responseBody)}`, { level: 'error', meta });
+          log(`API Error: ${processLogData(responseBody)}`, { level: 'error', meta });
```

Comment:
Suggestion: we could adopt the templating pattern used in react intl, so that the input data gets filtered automatically without having to call `processLogData` on the data object.

```suggestion
          log('API Error: $responseBody', { level: 'error', meta, data: { responseBody }});
```

Reply:
> the templating pattern used in react intl

Linky?

Reply:
I like the concept, and agree, BTW 🙂

Reply:
https://formatjs.io/docs/intl-messageformat#how-it-works 🚀

Reply:
What would we do for https://github.com/rewardops/rewardops-sdk-node/pull/70/files#diff-f387f6d23dddc249256731ec33c40cc8f36fa2850f5596222c1e19266e9cac14R124?

Reply:
One of the things we wanted to tackle in the overhaul and modernization is funnelling all calls through the `api` module, which would mitigate a fair amount of the headache here 🤔

Reply:
not sure how efficient the redacting and filtering functions are but, we only process the meta object if `verbose` is true

https://github.com/rewardops/rewardops-sdk-node/pull/70/files#diff-d2f66bef468d3d0209c80f15a385c9be3a2058b80ddf3612d13e779550faf819R284